lib/ansible/plugins/callback/unixy.py

```diff
@@ -131,6 +159,8 @@ def v2_runner_on_ok(self, result, msg="ok", display_color=C.COLOR_OK):
         if result_was_changed:
             msg = "done"
             display_color = C.COLOR_CHANGED
+        elif not self.display_ok_hosts:
```

Comment:
I think it'd be clearer if this "elif" were inverted, the return dropped, and the following lines made conditional as appropriate.